Automattic/wp-calypso
# Desktop App: fix crash on Apple Silicon caused by 8.0.0.

Fixes https://github.com/Automattic/wp-calypso/issues/78884.

## Proposed Changes

This PR adds fixes to the Entitlements list that caused the Apple Silicon build of the app to either crash or display a white screen on startup.

This PR also adds a fix for the failing Window build.

## What went wrong?

The previous release of WordPress.com Desktop App 8.0.0 caused crashes on users on the Apple Silicon architecture.

After almost a week of investigations, debugging and trial I found the culprit is because we jumped from Electron 12.x.x to Electron 24.x.x.

More specifically, [Electron has required (since 20.x.x)](https://github.com/electron/electron/issues/35355#issuecomment-1227574710) that additional entitlement be added in the form of `com.apple.security.cs.allow-jit`. 

This issue was not caught in https://github.com/Automattic/wp-calypso/pull/78098, because:
1. CircleCI builds were all successful;
2. locally built Apple Silicon builds launched as expected, and it was assumed that CI-produced builds will be the same;

As it turns out, the CI-produced builds involve the Apple signing process, which is where the Entitlements come into play. Because the Entitlements didn't change from Electron 12.x.x, when the binary was signed additional privileges required to run Electron >= 20.x.x was not granted, hence the app crashed on all Apple Silicon users.



## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] CircleCI > wp-desktop-mac
  - [ ] CircleCI > wp-desktop-linux
  - [ ] CircleCI > wp-desktop-windows

Additionally, download the x64 and arm64 produced binaries from CircleCI and ensure it functions as expected.

## Pre-merge Checklist
